ci-operator/step-registry/ipi/conf/alibabacloud/cloud-creds-deprovision/ipi-conf-alibabacloud-cloud-creds-deprovision-commands.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+
+CR_PATH=${SHARED_DIR}/crs
+
+export ALIBABA_CLOUD_CREDENTIALS_FILE=${SHARED_DIR}/inicreds
+cluster_id=${NAMESPACE}-${JOB_NAME_HASH}
+
+# extract ccoctl from the release image
+CCO_IMAGE=$(oc adm release info --image-for='cloud-credential-operator' "$RELEASE_IMAGE_LATEST")
+cd "/tmp"
+oc image extract "$CCO_IMAGE" --file="/usr/bin/ccoctl"
+chmod 555 "/tmp/ccoctl"
+
+mkdir -p ${CR_PATH}
+cp ${SHARED_DIR}/CREDENTIALSREQUEST_* ${CR_PATH}/
```

Comment:
It's weird to me that you're copying content into `${SHARED_DIR}` here.  Are there any later consumers?  If not, can `${CR_PATH}` be a directory in `/tmp/...` or some such, to avoid pushing into the cluster Secret things that aren't necessary?

Reply:
I agree this is strange. As you discovered later on, the consumer of this is the `ccoctl` command. The `ccoctl` for alibaba requires the credentials-requests. We could copy them (like we are doing) or we could extract them again. I chose to copy them. I believe the reason we need to keep them around / regenerate them is that the users and policies do not support tags. The clusterid could be used but the implementation wasn't done that way.